optuna_dashboard/_app.py

```diff
@@ -449,6 +452,48 @@ def save_trial_note(study_id: int, trial_id: int) -> dict[str, Any]:
         response.status = 204  # No content
         return {}
 
+    @app.get("/csv/<study_id:int>")
+    def download_csv(study_id: int) -> BottleViewReturn:
+        # Create a CSV file
+        summary = get_study_summary(storage, study_id)
+        if summary is None:
+            response.status = 404  # Not found
+            return {"reason": f"study_id={study_id} is not found"}
+        trials = get_trials(storage, study_id)
+        if len(trials) == 0:
+            return {"reason": f"study_id={study_id} has no trials"}
+
+        param_names = sorted(set(chain.from_iterable([t.params.keys() for t in trials])))
+        user_attr_names = sorted(set(chain.from_iterable([t.user_attrs.keys() for t in trials])))
+
+        param_names_header = [f"Param {x}" for x in param_names]
+        user_attr_names_header = [f"UserAttribute {x}" for x in user_attr_names]
+        value_header = ["Value"]
+        n_objs = max([len(t.values) for t in trials if t.values is not None], default=1)
```

Comment:
I think we need check the `study.directions` for the number of objectives.

Reply:
And, `len(study.metric_names)` if set.

Reply:
I see. I will revise the code, using `summary._system_attrs["study:metric_names"]` instead of `study.metric_names`.
```
            if "study:metric_names" in summary._system_attrs:
                value_header = summary._system_attrs["study:metric_names"]
```

Reply:
Finally, I obtained `study` and `study_name` from storage and avoid touching `_system_attrs`.